python/src/aiconfig/editor/client/src/hooks/useLoadModels.ts

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useEffect, useState } from "react";
+import { showNotification } from "@mantine/notifications";
+
+export default function useLoadModels(
+  modelSearch: string,
+  getModels: (search: string) => Promise<string[]>
+) {
+  const [models, setModels] = useState<string[]>([]);
+
+  const loadModels = useCallback(async (modelSearch: string) => {
+    try {
+      const models = await getModels(modelSearch);
+      setModels(models);
+    } catch (err: any) {
+      showNotification({
+        title: "Error loading models",
+        message: err?.message,
+        color: "red",
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    loadModels(modelSearch);
+  }, [loadModels, modelSearch]);
```

Comment:
How come we need to pass in `modelSearch` as a dependency here but not `getModels` as a dependency on L21? Is it because we're implicitly assuming that `getModels` should only be defined once and never change?

Reply:
Ah, good catch. `getModels` should be added above. Something must be up with my linter because it's missing these deps :(